openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -2236,6 +2215,118 @@ The following is a non-normative example of a Credential Request with Credential
 
 The value of the `credential` claim in the Credential Response MUST be a string that is an SD-JWT VC. Credentials of this format are already suitable for transfer and, therefore, they need not and MUST NOT be re-encoded.
 
+# Claims Description {#claims-description}
+
+A claims description object is an object used to describe how a certain claim in
+the Credential should be displayed to the End-User. It is used in the `claims`
+or `credentialSubject` parameters in the Credential Issuer metadata defined in
+(#format-profiles).
+
+For a claims description object, the following keys defined by this
+specification MAY be used to describe the claim or claims (other keys MAY also
+be used):
+
+  * `path`: REQUIRED. Array. Claim path query as defined in (#claims-path-query)
+    to identify the claim(s) in the Credential.
+  * `mandatory`: OPTIONAL. Boolean which, when set to `true`, indicates that the
+    Credential Issuer will always include this claim in the issued Credential.
+    If set to `false`, the claim is not included in the issued Credential if the
+    wallet did not request the inclusion of the claim, and/or if the Credential
+    Issuer chose to not include the claim. If the `mandatory` parameter is
+    omitted, the default value is `false`.
+  * `value_type`: OPTIONAL. String value determining the type of value of the
+    claim. Valid values defined by this specification are `string`, `number`,
+    and image media types such as `image/jpeg` as defined in IANA media type
+    registry for images
+    (https://www.iana.org/assignments/media-types/media-types.xhtml#image).
+    Other values MAY also be used.
+  * `display`: OPTIONAL. Array of objects, where each object contains display
+    properties of a certain claim in the Credential for a certain language.
+    Below is a non-exhaustive list of valid parameters that MAY be included:
+     * `name`: OPTIONAL. String value of a display name for the claim.
+     * `locale`: OPTIONAL. String value that identifies language of this object
+       represented as language tag values defined in BCP47 [@!RFC5646]. There
+       MUST be only one object for each language identifier.
+
+The order of claims description objects in the `claims` or `credentialSubject`
+array is used by the Wallet to determine the order in which the claims are
+displayed to the End-User, unless another mechanism is defined by the profile.
+
+# Claims Path Query {#claims-path-query}
+
+The metadata specifications above make use of claims path queries to select a
+particular claim in the credential or a set of such claims. This appendix
+defines the syntax and semantics of claims path queries.
+
+A claims path query MUST be a non-empty array of strings, `null` values, or
+non-negative integers. 
+
+A string indicates that the respective key is to be selected, a `null` value
```

Comment:
`null` -> `*`

would be more understandable ... but, before delving into these details, please consider the gap we have in IETF regarding the JSONPath (or similar approaches) and the requirement to get this standardized before propagating it in the openid specifications, in an unconventional way or in such a way that each specification, for specific needs, produces many narrow implementations with obvious implementation costs for the openid/ietf ecosystem.

As we know, JSONPath outlines a syntax for defining paths within a JSON document to find elements, much like XPath does for XML. It allows for filtering elements based on their attributes, navigating through arrays, and performing operations like slicing arrays.

Reply:
But `"*"` would be a string value which would require special rules for claims called `*` (admittedly, those shouldn't be too common).

JSONPath is a **huge** dependency. Suddenly, you have regular expressions and other stuff in your credential issuance protocol. Also some security issues, when you implement it incorrectly. And it brings so many things we don't need and would have to define-away...

Reply:
I don't consider myself a jsonpath lover, I'm wondering if there's the time to define a basic, common and general purpose approach for dealing with all these json handling. I agree to not use JsonPath. I think that we have to define an alternative way, not just tied in openid4vci, something to be reused, and expandend where necessarly.

Considering my concerns about the use of null in claims path queries being potentially misleading and carrying a negative connotation, I'm wondering on an alternative approach with a more intuitive and self-explanatory keyword that aligns with the intended functionality. 

we have "special" string like "*" or "".

Or using objects like `{"select": "all"}` that unfortunately carries more nesting and other concerns.

Reply:
I was thinking empty string as well instead of null (i think it's reasonable to not allow empty string names) but some libraries will probably also do weird stuff with an empty string.